### PR TITLE
Pass Bank::process_transactions() a reference instead of moving

### PR DESCRIPTION
**Pull request by garious**
When we move the transaction from input to `Result<Transaction>`, we lose the ability to reference the original transaction in the error case. Our choices are to populate the error structures with more data (specifically, the Signature) or change these functions to borrow the transaction. This PR chooses the latter. Not certain that's the right choice. Thoughts?

This PR is a WIP because it breaks the integration tests. Need to debug...

### CODE REVIEW

**Comment by rob-solana** [comment]
the goal of being able to reference the transaction in the error case is?  do we carry that error back to the program?

**Comment by garious** [comment]
@rob-solana, the goal of this change is to get to here: https://github.com/solana-labs/solana/pull/1301.

We're looking to store the error in a new lookup table where the key is the transaction's signature. There's other ways to achieve that, but usually idiomatic Rust code only uses move semantics to move data into structs, so I'm using this feature as an opportunity to stop using move semantics in process_transactions.  You see right out of the gate things look a little better because we can remove that `to_vec()`. Yes, we `clone()` the tx again later anyway, but at least it's pushed back. If we decide to store *all* transactions in the ledger (we can collect a fee on anything with a valid signature), we could delete that `clone()` too.

File: src/banking_stage.rs [hunk 1]
```diff
@@ -95,22 +95,22 @@ impl BankingStage {
...
-
+            let results = bank.process_transactions(&transactions[chunk_start..chunk_end]);
             debug!("results: {}", results.len());
 
             chunk_start = chunk_end;
 
             let mut hasher = Hasher::default();
 
-            let processed_transactions: Vec<_> = results
+            let processed_transactions: Vec<_> = transactions[chunk_start..chunk_end]
```
Comment (rob-solana) [inline src/banking_stage.rs]: move line 101 past this and you'll be golden

File: src/banking_stage.rs [hunk 2]
```diff
@@ -95,22 +95,22 @@ impl BankingStage {
         while chunk_start != transactions.len() {
             let chunk_end = chunk_start + Entry::num_will_fit(transactions[chunk_start..].to_vec());
 
-            let results = bank.process_transactions(transactions[chunk_start..chunk_end].to_vec());
-
+            let results = bank.process_transactions(&transactions[chunk_start..chunk_end]);
             debug!("results: {}", results.len());
 
             chunk_start = chunk_end;
```
Comment (rob-solana) [inline src/banking_stage.rs]: this line makes your processed_transactions vector empty....

**Pull request merged**